enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx

```diff
@@ -34,6 +37,16 @@ const MetabaseProviderInternal = ({
   theme,
 }: MetabaseProviderProps): JSX.Element => {
   const themeOverride = useMemo(() => {
+    if (theme?.colors) {
+      Object.entries(theme.colors).forEach(([key, value]) => {
+        colors[key as keyof MetabaseColors] = value;
+      });
+    } else {
+      Object.entries(originalColors).forEach(([key, value]) => {
+        colors[key as keyof MetabaseColors] = value;
+      });
+    }
```

Comment:
I wonder if we only pass in partial `theme.colors`, does the rest of the colors (i.e. `originalColors`) still gets applied? For instance, if I set `text-dark: '#2d2d30'` but didn't set anything else, does the default `mb-brand-color` gets applied?

Also, I think we are globally mutating the `colors` object here, so semantically I'm not sure if putting it in `useMemo` make sense, as `useMemo` usually communicates purity.

Reply:
@deniskaber Yeah, I couldn't think of a way to cleanly do this without an extra re-render cycle either, so I think this will have to do 👍🏻

Reply:
A way to make this logic jump out a bit more would be to move it to a named function, say `setGlobalMetabaseColors`. It kinda makes the logic jumps out more that it's not doing the usual thing that you would expect it to.